src/dao.ts

```diff
@@ -0,0 +1,113 @@
+import { BigInt, Address, log } from "@graphprotocol/graph-ts";
+import { LogVote } from "../generated/Dao/Dao";
+import { LogFundsDistributed, Dao } from "../generated/Dao/Dao";
+import {
+  VoteManager,
+  Iteration,
+  VoteStatus,
+  PatronNew,
+  Vote,
+} from "../generated/schema";
+import { VOTES_MANAGER_ENTITY_ID } from "./CONSTANTS";
+
+export function handleLogVote(event: LogVote): void {
+  let proposalVotedFor = event.params.proposalVotedFor;
+  let votesCast = event.params.votesCast;
+  let totalVotesForProposal = event.params.totalVotesForProposal;
+  let totalVotesAllProposals = event.params.totalVotesAllProposals;
+  let addressOfVoter = event.params.addressOfVoter;
+  let voteManagerContract = Dao.bind(event.address);
+  let currentIteration = voteManagerContract.proposalIteration();
+  let currentIterationString = currentIteration.toString();
+  let projectVoteId =
+    currentIterationString + "-" + proposalVotedFor.toString();
+
+  let iteration = Iteration.load(currentIterationString);
+  if (iteration == null) {
+    iteration = new Iteration(currentIterationString);
+    iteration.totalVotes = BigInt.fromI32(0);
+    iteration.winningProposal = BigInt.fromI32(0);
+    iteration.fundsDistributed = BigInt.fromI32(0);
+    iteration.winningVotes = BigInt.fromI32(0);
+    iteration.projectVoteTallies = [];
+    iteration.individualVotes = [];
+  }
+  iteration.totalVotes = totalVotesAllProposals;
+
+  // TODO: create event in the initialize function.
+  let voteManager = VoteManager.load(VOTES_MANAGER_ENTITY_ID);
+  if (voteManager == null) {
+    voteManager = new VoteManager(VOTES_MANAGER_ENTITY_ID);
+
+    voteManager.currentIteration = iteration.id;
```

Comment:
Yeah, maybe you need to save the iteration before this line. Or maybe you need to run the `voteManager.save()` right at the end?

Try with `log.warn` and see if you can work out why it isn't defined.

Reply:
I fixed it all. Also the voteManager now has the currentIteration (which is not complete and has votes etc...) and the lastCompleteIteration (which makes sense). Check out the changes